pkg/provider/cloud/baremetal/provider.go

```diff
@@ -32,11 +35,36 @@ func NewCloudProvider() provider.CloudProvider {
 
 var _ provider.CloudProvider = &baremetal{}
 
-func (b *baremetal) DefaultCloudSpec(_ context.Context, _ *kubermaticv1.ClusterSpec) error {
+func (b *baremetal) DefaultCloudSpec(_ context.Context, spec *kubermaticv1.ClusterSpec) error {
+	if spec.Cloud.Baremetal == nil {
+		return errors.New("baremetal cloud provider spec is empty")
+	}
+
+	if spec.Cloud.Baremetal != nil && spec.Cloud.Baremetal.Tinkerbell == nil {
+		return errors.New("tinkerbell spec is empty! we must have a provisioner")
+	}
 	return nil
 }
 
-func (b *baremetal) ValidateCloudSpec(_ context.Context, _ kubermaticv1.CloudSpec) error {
+func (b *baremetal) ValidateCloudSpec(_ context.Context, spec kubermaticv1.CloudSpec) error {
+	if spec.Baremetal.Tinkerbell != nil {
+		kubeconfig := spec.Baremetal.Tinkerbell.Kubeconfig
+		if kubeconfig == "" {
+			return errors.New("tinkerbell kubeconfig is not found")
+		}
+		config, err := base64.StdEncoding.DecodeString(kubeconfig)
+		if err != nil {
```

Comment:
Let's not introduce a "_can_ be base64, but _can_ also be plaintext" logic. Your own documentation on the API types say that this must be base64 encoded, so either that's the format or it isn't.

Over the years we had soooooooooooooo many (more than you can imagine) issues because nobody could remember where something had to be base64 encoded and where not, we should not introduce more uncertainty.

Reply:
I would personally even check if we can store the kubeconfig as a Secret, instead of base64'd inline'd in the Cluster object. Inline kubeconfigs are always bad and cumbersome.